django/db/backends/sqlite3/base.py

```diff
@@ -202,48 +203,51 @@ def get_connection_params(self):
-        conn.create_function("django_timestamp_diff", 2, _sqlite_timestamp_diff)
-        conn.create_function("django_format_dtdelta", 3, _sqlite_format_dtdelta)
-        conn.create_function('regexp', 2, _sqlite_regexp)
-        conn.create_function('ACOS', 1, none_guard(math.acos))
-        conn.create_function('ASIN', 1, none_guard(math.asin))
-        conn.create_function('ATAN', 1, none_guard(math.atan))
-        conn.create_function('ATAN2', 2, none_guard(math.atan2))
-        conn.create_function('CEILING', 1, none_guard(math.ceil))
-        conn.create_function('COS', 1, none_guard(math.cos))
-        conn.create_function('COT', 1, none_guard(lambda x: 1 / math.tan(x)))
-        conn.create_function('DEGREES', 1, none_guard(math.degrees))
-        conn.create_function('EXP', 1, none_guard(math.exp))
-        conn.create_function('FLOOR', 1, none_guard(math.floor))
-        conn.create_function('LN', 1, none_guard(math.log))
-        conn.create_function('LOG', 2, none_guard(lambda x, y: math.log(y, x)))
-        conn.create_function('LPAD', 3, _sqlite_lpad)
-        conn.create_function('MD5', 1, none_guard(lambda x: hashlib.md5(x.encode()).hexdigest()))
-        conn.create_function('MOD', 2, none_guard(math.fmod))
-        conn.create_function('PI', 0, lambda: math.pi)
-        conn.create_function('POWER', 2, none_guard(operator.pow))
-        conn.create_function('RADIANS', 1, none_guard(math.radians))
-        conn.create_function('REPEAT', 2, none_guard(operator.mul))
-        conn.create_function('REVERSE', 1, none_guard(lambda x: x[::-1]))
-        conn.create_function('RPAD', 3, _sqlite_rpad)
-        conn.create_function('SHA1', 1, none_guard(lambda x: hashlib.sha1(x.encode()).hexdigest()))
-        conn.create_function('SHA224', 1, none_guard(lambda x: hashlib.sha224(x.encode()).hexdigest()))
-        conn.create_function('SHA256', 1, none_guard(lambda x: hashlib.sha256(x.encode()).hexdigest()))
-        conn.create_function('SHA384', 1, none_guard(lambda x: hashlib.sha384(x.encode()).hexdigest()))
-        conn.create_function('SHA512', 1, none_guard(lambda x: hashlib.sha512(x.encode()).hexdigest()))
-        conn.create_function('SIGN', 1, none_guard(lambda x: (x > 0) - (x < 0)))
-        conn.create_function('SIN', 1, none_guard(math.sin))
-        conn.create_function('SQRT', 1, none_guard(math.sqrt))
-        conn.create_function('TAN', 1, none_guard(math.tan))
+        create_deterministic_function('django_time_extract', 2, _sqlite_time_extract)
+        create_deterministic_function('django_time_trunc', 2, _sqlite_time_trunc)
+        create_deterministic_function('django_time_diff', 2, _sqlite_time_diff)
+        create_deterministic_function('django_timestamp_diff', 2, _sqlite_timestamp_diff)
+        create_deterministic_function('django_format_dtdelta', 3, _sqlite_format_dtdelta)
+        create_deterministic_function('regexp', 2, _sqlite_regexp)
```

Comment:
I theory a locale dependant pattern could be provided and `locale.setlocale` be called but this is really unlikely.